wandb/sdk/wandb_run.py

```diff
@@ -1273,39 +1287,29 @@ def _log(
         if any(not isinstance(key, str) for key in data.keys()):
             raise ValueError("Key values passed to `wandb.log` must be strings.")
 
+        self._partial_history_callback(data, step, commit)
+
         if step is not None:
+            if os.getpid() != self._init_pid or self._is_attached:
```

Comment:
Thinking out loud: I remember seeing something about some weird behavior of `os.getpid` on Windows, but can't find it...

Reply:
Found this: https://stackoverflow.com/questions/46058118/why-os-getpid-returns-a-negative-pid-on-windows

Reply:
I guess the only issue is if we unpickle on a different platform, but this seems like a harder things to support...

Reply:
ok marking as resolved but it is a good point to keep in mind... maybe getpid is not robust enough